controllers/etcd_copy_backups_controller.go

```diff
@@ -0,0 +1,446 @@
+		var job *batchv1.Job
+		if err != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+		if job, err = r.runCopyJob(ctx, copyTask, values, logger); err != nil {
+			if statusError := r.updateEtcdCopyTaskStatusLastError(ctx, copyTask, err); statusError != nil {
+				return ctrl.Result{}, fmt.Errorf("%s; %s", err, statusError)
+			}
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+		if err := r.addFinalizerToObject(ctx, job, logger); err != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+		if err := r.updateEtcdCopyTaskStatusWithTaskDescription(ctx, copyTask, job.Name, reflect.TypeOf(job).Elem().Name(), batchv1.SchemeGroupVersion.String()); err != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+	}
+
+	if err := r.updateEtcdCopyBackupsTaskStatus(ctx, copyTask); err != nil {
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: 30 * time.Second,
+	}, nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) delete(ctx context.Context, copyTask *druidv1alpha1.EtcdCopyBackupsTask) (ctrl.Result, error) {
+	logger := r.logger.WithValues("etcd", kutil.Key(copyTask.Namespace, copyTask.Name).String(), "operation", "delete")
+	logger.Info("Starting operation")
+
+	if copyTask.Status.TaskReference != nil {
+		if err := r.updateEtcdCopyBackupsTaskStatus(ctx, copyTask); err != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+		if copyTask.Status.LastOperation != nil && copyTask.Status.LastOperation.State == druidv1alpha1.LastOperationStateActive {
+			return ctrl.Result{
+				Requeue: true,
+			}, errors.New("The resource can not be deleted while still processing")
+		}
+		if !reflect.DeepEqual(copyTask.Status.TaskReference.Kind, "") && !reflect.DeepEqual(copyTask.Status.TaskReference.Kind, "Job") {
+			return ctrl.Result{
+				Requeue: true,
+			}, fmt.Errorf("Unsupported task kind %s", copyTask.Status.TaskReference.Kind)
+		}
+
+		job := &batchv1.Job{}
+		if err := r.Get(ctx, kutil.Key(copyTask.Namespace, copyTask.Status.TaskReference.Name), job); err == nil {
+			if err := r.removeFinalizerFromObject(ctx, job, logger); err != nil {
+				return ctrl.Result{
+					Requeue: true,
+				}, err
+			}
+			if err := r.Delete(ctx, job); client.IgnoreNotFound(err) != nil {
+				return ctrl.Result{
+					Requeue: true,
+				}, err
+			}
+		} else if client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{
+				Requeue: true,
+			}, err
+		}
+	}
+	if err := r.removeFinalizerFromObject(ctx, copyTask, logger); err != nil {
+		return ctrl.Result{
+			Requeue: true,
+		}, err
+	}
+
+	logger.Info(fmt.Sprintf("Deleted ETCDCopyBackupTask %s/%s successfully.", copyTask.Namespace, copyTask.Name))
+	return ctrl.Result{}, nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) updateEtcdCopyTaskStatusWithTaskDescription(ctx context.Context, copyTask *druidv1alpha1.EtcdCopyBackupsTask, name, kind, apiGroup string) error {
```

Comment:
I think there are too many different status update methods. Could you try to combine at least `updateEtcdCopyTaskStatusWithTaskDescription` and `updateEtcdCopyBackupsTaskStatus` into one method? There is no need to pass `name`, `kind`, and `apiGroup` as parameters here from my perspective, you can hardcode / compute these already in the method.